### ISSUE TITLE
Guava version at bblfsh/scala-client

### ISSUE THREAD

**Issue by bzz** [src-d/engine#142]
On clean Ubuntu 16.04 env using

`bin/spark-shell --packages tech.sourced:engine:0.1.2 --repositories "https://jitpack.io"`
`--repositories` is needed for siva-java, not included in uber-jar 0.1.2
and

Example \w [UAST extraction](https://github.com/src-d/engine/blob/master/_examples/scala/spark-shell-xpath-query.md) I got Guava version mismatch at runtime

```
17/10/26 13:59:37 WARN TaskSetManager: Lost task 0.0 in stage 1.0 (TID 10, localhost, executor driver): java.lang.NoSuchMethodError: com.google.common.util.concurre$
t.MoreExecutors.directExecutor()Ljava/util/concurrent/Executor;
        at io.grpc.internal.ClientCallImpl.<init>(ClientCallImpl.java:104)
        at io.grpc.internal.ManagedChannelImpl$RealChannel.newCall(ManagedChannelImpl.java:554)
        at io.grpc.internal.ManagedChannelImpl.newCall(ManagedChannelImpl.java:533)
        at gopkg.in.bblfsh.sdk.v1.protocol.generated.ProtocolServiceGrpc$ProtocolServiceBlockingStub.parse(ProtocolServiceGrpc.scala:50)
        at org.bblfsh.client.BblfshClient.parse(BblfshClient.scala:29)
        at tech.sourced.engine.udf.ExtractUASTsUDF$.extractUsingBblfsh(ExtractUASTsUDF.scala:110)
        at tech.sourced.engine.udf.ExtractUASTsUDF$.extractUAST(ExtractUASTsUDF.scala:92)
        at tech.sourced.engine.udf.ExtractUASTsUDF$.extractUASTsWithLang(ExtractUASTsUDF.scala:69)
```

**Comment by ajnavarro** [comment]
Should be fixed on 0.1.3

**Comment by bzz** [comment]
I see!

Please correct me if i'm wrong, but does that mean that we could release 0.1.4 and then ` --packages tech.sourced:engine:0.1.4` would work, and not require `--repositories` ?

**Comment by EgorBu** [comment]
Thanks @bzz , does anyone know when it should be released?

**Comment by erizocosmico** [comment]
No, it means if we release 0.1.4 there will be a fatjar in the github release (not the maven one). For that to work with packages fatjar needs to be in maven, which we don't do yet (but we can totally do)

**Comment by bzz** [comment]
I see, thank you for explaining! 

@erizocosmico  @ajnavarro what do you guys think, shall we change the release process to publish a fatjat and make 0.1.4 then? I think that would solve this issue and unblock @EgorBu

**Comment by erizocosmico** [comment]
Ok, will make the change and send the PR

**Comment by ajnavarro** [comment]
Yep, having the guava shading (and I don't remember if we are shading netty too) is not possible to run the engine jar without make it fat. So is totally a good idea to upload to maven the fatjar (but not with all spark inside, just with the needed dependencies; enry, bblfsh, etc).

**Comment by bzz** [comment]
Should be fixed by #143

**Comment by bzz** [comment]
Closing, as #143 merged, 0.1.4 has landed on Maven central https://search.maven.org/#artifactdetails%7Ctech.sourced%7Cengine%7C0.1.4%7Cjar and it works https://github.com/src-d/backlog/issues/1089#issuecomment-339935927
